packager/mpd/base/xml/xml_node.h

```diff
@@ -66,6 +66,10 @@ class XmlNode {
   /// @param id is the ID for this element.
   void SetId(uint32_t id);
 
+  /// Sets 'id=rep_id' attribute.
+  /// @param id is the ID for this element.
+  void SetIdString(std::string id);
```

Comment:
this is a problem for the same reason I mentioned elsewhere, even internally id is expected to be uint. So the option of passing in an alternate string ID isn't great.